lib/rubocop/cop/util.rb

```diff
@@ -212,32 +212,32 @@ def double_quotes_required?(string)
 
         # Regex matches IF there is a ' or there is a \\ in the string that is
         # not preceded/followed by another \\ (e.g. "\\x34") but not "\\\\".
-        string.inspect =~ /'|(?<! \\) \\{2}* \\ (?![\\"])/x
+        string =~ /'|(?<! \\) \\{2}* \\ (?![\\"])/x
       end
 
       # If double quoted string literals are found in Ruby code, and they are
       # not the preferred style, should they be flagged?
       def double_quotes_acceptable?(string)
         # If a string literal contains hard-to-type characters which would
         # not appear on a "normal" keyboard, then double-quotes are acceptable
-        double_quotes_required?(string) ||
+        needs_escaping?(string) ||
           string.codepoints.any? { |cp| cp < 32 || cp > 126 }
       end
 
+      def needs_escaping?(string)
+        double_quotes_required?(string.inspect)
```

Comment:
I ended up using that "escape" method because it was already there: https://github.com/bbatsov/rubocop/pull/3189/commits/44af269942180048355f18900126e5745de2b27f :)